integration/client/apps/apps_test.go

```diff
@@ -461,3 +462,37 @@ func TestAppRun(t *testing.T) {
 	assert.Equal(t, "secret", app.Spec.Secrets[0].Secret)
 	assert.Equal(t, "value", app.Spec.DeployArgs["key"])
 }
+
+func TestAppRunImageVariations(t *testing.T) {
+	helper.EnsureCRDs(t)
+	restConfig := helper.StartAPI(t)
+
+	ctx := helper.GetCTX(t)
+	kclient := helper.MustReturn(kclient.Default)
+	ns := helper.TempNamespace(t, kclient)
+
+	imageID := client2.NewImage(t, ns.Name)
+
+	c, err := client.New(restConfig, "", ns.Name)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := c.ImageTag(ctx, imageID, "foo/bar:baz"); err != nil {
+		t.Fatal(err)
+	}
+
+	imageNames := []string{
+		"foo/bar:baz",
+		imageID,
+		fmt.Sprintf("sha256:%s", imageID),
+		imageID[:8],
+	}
+
+	for _, imageName := range imageNames {
+		_, err := c.AppRun(ctx, imageName, nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+}
```

Comment:
nit: I wonder if we could use sub tests to parallelize these test cases

```suggestion
func TestAppRunImageVariations(t *testing.T) {
	helper.EnsureCRDs(t)
	restConfig := helper.StartAPI(t)

	ctx := helper.GetCTX(t)
	kclient := helper.MustReturn(kclient.Default)
	ns := helper.TempNamespace(t, kclient)

	imageID := client2.NewImage(t, ns.Name)

	c, err := client.New(restConfig, "", ns.Name)
	if err != nil {
		t.Fatal(err)
	}

	if err := c.ImageTag(ctx, imageID, "foo/bar:baz"); err != nil {
		t.Fatal(err)
	}

    for desc, imageName := range map[string]string{
        "ref": "foo/bar:baz",
        "id": imageID,
        "sha256": fmt.Sprintf("sha256:%s", imageID),
        "short": imageID[:8],
    } {
        t.Run(desc, func(t *testing.T) {
             t.Parallel()
             _, err := c.AppRun(ctx, imageName, nil)
             assert.NoError(t, err)
        }
    }
}
```

Reply:
https://github.com/acorn-io/acorn/pull/1731